deploy/charts/rook-ceph/templates/deployment.yaml

```diff
@@ -31,10 +31,7 @@ spec:
         image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
         imagePullPolicy: {{ .Values.image.pullPolicy }}
         args: ["ceph", "operator"]
-        securityContext:
-          runAsNonRoot: true
-          runAsUser: 2016
-          runAsGroup: 2016
+        securityContext: {{- .Values.containerSecurityContext | toYaml | nindent 10 }}
```

Comment:
There are more settings in the security context than just the 4 specified now (allow privilege escalation, privileged, read only file system, seccomp profiles and the upcoming appArmor). If the context is customizable this allows users to additionally specify these settings for whatever custom environment rook runs in. 

I think #11755 can't be fixed as the hostpath volumes are required unless rook is provisioned with pvc storage instead of host disks.